doc/source/user/basics.creation.rst

```diff
@@ -9,138 +9,353 @@ Array creation
+    signed c: [-3 -3 -3] int64
+
+Notice when you perform operations with two arrays of the same
+``dtype``: ``uint32``, the resulting array is the same type. When you
+perform operations with different ``dtype``, NumPy will 
+assign a new type that satisfies all of the array elements involved in
+the computation, here ``uint32`` and ``int32`` can both be represented in
+as ``int64``. 
+
+The default NumPy behavior is to create arrays in either 64-bit signed
+integers or double precision floating point numbers, ``int64`` and
+``float``, respectively. If you expect your arrays to be a certain type,
+then you need to specify the ``dtype`` while you create the array. 
+
+2) Intrinsic NumPy array creation objects
+=========================================
+..
+  40 functions seems like a small number, but the routies.array-creation
+  has ~47. I'm sure there are more. 
+
+NumPy has over 40 built-in functions for creating arrays as laid
+out in the :ref:`Array creation routines <routines.array-creation>`.
+There are three categories of NumPy array creation objects:
+1) 1D arrays
+2) 2D arrays
+3) ND arrays
+
+1 - 1D array creation objects
+-----------------------------
+
+The 1D array creation objects e.g. :func:`numpy.linspace` and
+:func:`numpy.arange` generally  need at least two inputs ``start`` and
+``stop``. Where ``start`` is the first value in the array and ``stop``
+specifies the value of the last element in the array. 
+
+:func:`numpy.arange` creates arrays with regularly incrementing values.
+Check the docstring for complete information and examples. A few
+examples are shown: ::
+
+ >>> arange(10)
  array([0, 1, 2, 3, 4, 5, 6, 7, 8, 9])
- >>> np.arange(2, 10, dtype=float)
+ >>> arange(2, 10, dtype=float)
  array([ 2., 3., 4., 5., 6., 7., 8., 9.])
- >>> np.arange(2, 3, 0.1)
+ >>> arange(2, 3, 0.1)
  array([ 2. , 2.1, 2.2, 2.3, 2.4, 2.5, 2.6, 2.7, 2.8, 2.9])
 
-Note that there are some subtleties regarding the last usage that the user
-should be aware of that are described in the arange docstring.
+Note: there are some subtleties regarding ``dtype``. In the second
+example, the ``dtype`` is defined. In the third example, the array is
+``dtype = float`` to accomodate the step size of ``0.1``. 
 
-linspace() will create arrays with a specified number of elements, and
+:func:`numpy.linspace` will create arrays with a specified number of elements, and
 spaced equally between the specified beginning and end values. For
 example: ::
 
- >>> np.linspace(1., 4., 6)
+ >>> linspace(1., 4., 6)
  array([ 1. ,  1.6,  2.2,  2.8,  3.4,  4. ])
 
-The advantage of this creation function is that one can guarantee the
-number of elements and the starting and end point, which arange()
-generally will not do for arbitrary start, stop, and step values.
+The advantage of this creation object is that you guarantee the
+number of elements and the starting and end point. The previous
+``arange(start, stop, step)`` will not include the value ``stop``.
```

Comment:
Any reason to change from function to object in the text above?

I think it might be helpful to include a note here about [expected `arange` behaviour](https://github.com/numpy/numpy/issues/17342) with non-integer steps.

Reply:
> Any reason to change from function to object in the text above?

I missed that mention of "function". I originally wrote all my notes as referring to "functions" instead of the preexisting vocabulary "objects".

I'll change ~~function~~ -> "object"

> I think it might be helpful to include a note here about [expected `arange` behaviour](https://github.com/numpy/numpy/issues/17342) with non-integer steps.

I hadn't run into this problem before, thanks for suggesting the note! This is a great place to discuss this behavior. 

Why does `arange` sometimes include the `stop` value? Is it a roundoff issue?

Reply:
> Any reason to change from function to object in the text above?

I think the question was the opposite: use `function` not `object`. Although everything in python is an object, it is more precise to refer to the routines as `functions`.

> Why does arange sometimes include the stop value? Is it a roundoff issue?

Yes, and it is a common source of new user confusion.